src/index.js

```diff
@@ -49,9 +50,13 @@ class CID {
    * new CID(<cid.buffer>)
    * new CID(<multihash>)
    * new CID(<bs58 encoded multihash>)
+   * new CID(<CID>)
    *
    */
   constructor (version, codec, multihash) {
+    if (CID.isCID(version)) {
+      return version
```

Comment:
If you call `new CID` using an already created CID, actually, it raises an error.  
In this way, it should give you the CID back.

```js
cid = new CID(...)
other_cid = new CID(cid) // other_cid === cid
```

I thought it could be useful, to avoid to check and clean the cid (e.g. [here](https://github.com/ipfs/js-ipfs/blob/master/src/core/components/block.js#L76), used [here](https://github.com/ipfs/js-ipfs/blob/master/src/core/components/block.js#L12))